### PR TITLE
Make ResultCache max_size configurable

### LINKED ISSUE #463: Limit of 100001 for URLs
## Summary
The total URLs has a limit of **100001**, reached this limit the value remains unchanged even if the scan proceeds and "_links queued_" and "_links checked_" reach values of millions of units:

100 threads active, 30673 links queued, 68486 links in 99903 URLs checked, runtime 2 hours, 43 minutes
100 threads active, 30698 links queued, 68541 links in 99984 URLs checked, runtime 2 hours, 43 minutes
100 threads active, 30720 links queued, 68636 links in **100001** URLs checked, runtime 2 hours, 43 minutes
100 threads active, 30748 links queued, 68706 links in **100001** URLs checked, runtime 2 hours, 43 minutes
100 threads active, 30776 links queued, 68776 links in **100001** URLs checked, runtime 2 hours, 43 minutes
100 threads active, 30800 links queued, 68836 links in **100001** URLs checked, runtime 2 hours, 44 minutes
[...]
100 threads active, 168972 links queued, 223066 links in **100001** URLs checked, runtime 7 hours, 1 minute
100 threads active, 169039 links queued, 223083 links in **100001** URLs checked, runtime 7 hours, 1 minute
[...]
100 threads active, 2489377 links queued, 1461870 links in **100001** URLs checked, runtime 1 day, 12 hours, 21 minutes
100 threads active, 2489509 links queued, 1461942 links in **100001** URLs checked, runtime 1 day, 12 hours, 21 minutes

This bug is independent of the number of threads, occurs with either 1, 10, 100 or 1000

1 thread active, 2836180 links queued, 1576480 links in **100001** URLs checked, runtime 5 days, 23 hours, 6 minutes
10 threads active, 5415569 links queued, 3007833 links in **100001** URLs checked, runtime 3 days, 8 hours, 27 minutes
100 threads active, 2489509 links queued, 1461942 links in **100001** URLs checked, runtime 1 day, 12 hours, 21 minutes
1000 threads active, 688499 links queued, 629969 links in **100001** URLs checked, runtime 23 hours, 47 minutes

## Steps to reproduce

 1. Run linkchecker on a site with more than 100001 URLs

## Actual result

10 threads active, 5415569 links queued, 3007833 links in **100001** URLs checked, runtime 3 days, 8 hours, 27 minutes

## Expected result

something like:
10 threads active, 5415569 links queued, 3007833 links in **8423402** URLs checked, runtime 3 days, 8 hours, 27 minutes

## Environment

 * Operating system:
LSB Version:	:base-4.0-amd64:base-4.0-noarch:core-4.0-amd64:core-4.0-noarch:graphics-4.0-amd64:graphics-4.0-noarch:printing-4.0-amd64:printing-4.0-noarch
Distributor ID:	RedHatEnterpriseServer
Description:	Red Hat Enterprise Linux Server release 6.10 (Santiago)
Release:	6.10
Codename:	Santiago
 * Linkchecker version: 
WARNING linkcheck.check 2020-07-29 12:39:48,165 MainThread Running as root user; dropping privileges by changing user to nobody.
LinkChecker 10.0.0.dev0 released xx.xx.xxxx
Copyright (C) 2000-2014 Bastian Kleineidam
 * Python version: Python 3.6.3
 * Install method: 
pip install --verbose --upgrade https://github.com/linkchecker/linkchecker/archive/master.zip
 * Site URL: Every site win 100001+ URLs or the site already sent with a private message to @mgedmin for the #65 

Regards

  Mirco

### PR DESCRIPTION
**Pull request by phaer**
fixes #463

### CODE REVIEW

**Review by cjmayo — CHANGES_REQUESTED** [review]
man pages are generated from `doc/src/man/linkcheckerrc.rst`. Please update that file and run the man page generation as documented in `doc/documentation.md`.

**Comment by phaer** [comment]
Ah, thanks for the hint and sorry, been too sloppy w.r.t. the documentation. Hope that fixes it :)

File: doc/src/man/linkcheckerrc.rst [hunk 1]
```diff
@@ -95,9 +95,13 @@ checking
     be accessed before checking.
     The default is to use robots.txt files.
     Command line option: :option:`--no-robots`
-**allowedschemes=**\ *NAME*\ [**,**\ *NAME*...]
+**allowedschemes=**\ *NAME*\ [**,**\ *NAME*...**
```
Comment (cjmayo) [inline doc/src/man/linkcheckerrc.rst:98]: `**` is a typo? Would be a different change anyway.

**Review by cjmayo — CHANGES_REQUESTED** [review]

File: doc/src/man/linkcheckerrc.rst (hunk 1 above)
Comment (phaer) [reply doc/src/man/linkcheckerrc.rst]: Yes, it is. Thanks for noticing.

**Review by cjmayo — APPROVED** [review]
OK good to go. Thanks for replacing a magic number with a configurable setting!

**Pull request merged**